alert/tests.py

```diff
@@ -532,21 +534,341 @@ def test_no_permission(self):
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], None)
+        self.assertEqual(response.data['profile']['phone'], None)
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_multiple_users_independent(self):
+        User.objects.create_user(username='murey',
+                                 password='top_secret')
+        client2 = APIClient()
+        client2.login(username='murey', password='top_secret')
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob',
+                                               'profile': {'email': 'example@email.com', 'phone': '3131234567'}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = client2.put('/api/settings/',
+                               json.dumps({'username': 'murey',
+                                           'profile': {'email': 'example2@email.com', 'phone': '2121234567'}}),
+                               content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example2@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+12121234567')
+        self.assertEqual(response.data['username'], 'murey')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = client2.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example2@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+12121234567')
+        self.assertEqual(response.data['username'], 'murey')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
```

Comment:
Should add another `TestCase` for Registrations. Some ideas for tests that would be good to have:
- Test to make sure we only return registrations for the current semester
- Test for successfully submitting registrations for a given user
- Test for trying to submit a registration for another user